tsl/src/bgw_policy/job.c

```diff
@@ -250,9 +250,50 @@ bool
 policy_retention_execute(int32 job_id, Jsonb *config)
 {
 	PolicyRetentionData policy_data;
+	char *relname;
+	char *message = "applying retention policy to hypertable";
+	Datum boundary;
 
 	policy_retention_read_and_validate_config(config, &policy_data);
 
+	relname = get_rel_name(policy_data.object_relid);
+	boundary = policy_data.boundary;
+	if (IS_INTEGER_TYPE(policy_data.boundary_type))
+	{
+		elog(LOG,
+			 "%s: %s dropping data older than %s",
+			 message,
+			 relname,
+			 DatumGetCString(DirectFunctionCall1(int8out, boundary)));
+	}
+	switch (policy_data.boundary_type)
+	{
+		case TIMESTAMPOID:
+			elog(LOG,
+				 "%s: \"%s\" dropping data older than %s",
+				 message,
+				 relname,
+				 DatumGetCString(DirectFunctionCall1(timestamp_out, boundary)));
+			break;
+		case TIMESTAMPTZOID:
+			elog(LOG,
+				 "%s: \"%s\" dropping data older than %s",
+				 message,
+				 relname,
+				 DatumGetCString(DirectFunctionCall1(timestamptz_out, boundary)));
+			break;
+		case DATEOID:
+			elog(LOG,
+				 "%s: \"%s\" dropping data older than %s",
+				 message,
+				 relname,
+				 DatumGetCString(DirectFunctionCall1(date_out, boundary)));
+			break;
+		default:
+			/* this should never happen as otherwise hypertable has unsupported time type */
```

Comment:
If it's truely impossible, you should add an `Assert()` here. Otherwise, you could add a print or throw an exception.

Reply:
How about an `ereport` with `ERRCODE_INVALID_PARAMETER_VALUE` ?